src/components/search.js

```diff
@@ -47,6 +77,13 @@ export default class Search extends React.PureComponent {
           placeholder={i18n.search}
           autoFocus={autoFocus}
         />
+        <div
+          className="emoji-mart-search-icon"
+          onClick={this.clear}
+          tabIndex={isSearching ? '0' : ''}
+          onKeyUp={this.handleKeyUp}
```

Comment:
AFAIK If you change this to be a `<button>` instead of a `<div>` it would auto add the handles for space and enter on keyup and you wouldn't have to listen for them it was simply fire the onClick automagically

Also this would remove the need for a `tabIndex` as I was reading up on tabIndex there are some negatives to forcing a `tabIndex` to make the div tab-able https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/tabindex#Accessibility_concerns

Which would then remove the need for `isSearching` on the state

Reply:
So I went ahead and tried this, and found that it auto adds the handler for space but not for enter on keyup so I needed to keep keyup in there to handle the enter case. The tabIndex can now be removed but in order to keep the search icon disabled (and the delete icon enabled), I need `isSearching` to toggle the `disabled` attribute. 

Since the `tabIndex` gets avoided with this implementation and it's a bit cleaner than the `<div>`, I'll go ahead and make the change!